azure-devops/azext_devops/dev/common/github_credential_manager.py

```diff
@@ -0,0 +1,79 @@
+# --------------------------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for license information.
+# --------------------------------------------------------------------------------------------
+
+import base64
+import requests
+from knack.prompting import prompt, prompt_pass
+from knack.log import get_logger
+from knack.util import CLIError
+from azext_devops.dev.common.utils import randomword, singleton
+
+logger = get_logger(__name__)
+
+
+@singleton
+class GithubCredentialManager():
+    """ GithubCredentialManager
+    """
+    def __init__(self):
+        self.username = None
```

Comment:
are these really needed as class level vars?

Reply:
Yes, this is a singleton class so if the credentials are needed again during the command run we reuse the token from the class var.